CBCCasper.thy

```diff
@@ -0,0 +1,160 @@
+theory CBCCasper
+
+imports Main HOL.Real
+
+begin
+
+(* Section 2: Description of CBC Casper *)
+(* Section 2.1: CBC Casper "Parameters" *)
+
+notation Set.empty ("\<emptyset>")
+
+definition is_non_empty :: "'a set \<Rightarrow> bool"
+  where
+    "is_non_empty bs = ( \<exists> b. b \<in> bs )"
+
+(* Definition 2.1 ~ 2.5 *)
+datatype validator = Validator int
```

Comment:
`Validator real` ?

Reply:
The data type of `validator names` is not defined in the paper. I prefer `int` to represent IDs of validators.